test/runner/runner.js

```diff
@@ -214,4 +232,30 @@ class Worker extends EventEmitter {
   }
 }
 
+function collectRequires(file, allDeps) {
+  if (allDeps.has(file))
+    return;
+  allDeps.add(file);
+  const cache = require.cache[file];
+  const deps = cache.children.map(m => m.id);
+  for (const dep of deps)
+    collectRequires(dep, allDeps);
+}
+
+function computeWorkerHash(file) {
+  // At this point, filesWithRegistrations contains all the files with worker fixture registrations.
+  // For every test, build the require closure and map each file to fixtures declared in it.
+  // This collection of fixtures is the fingerprint of the worker setup, a "worker hash".
+  // Tests with the matching "worker hash" will reuse the same worker.
```

Comment:
If I have a `server` worker fixture, and I add an unrelated `pokemon` worker fixture in my test, does this mean I will get a new server instance?

Reply:
Yes, you will. I can trace fixture usages and overrides to mitigate this.

Reply:
Having said that, I am not sure it is critical - all I did was I recycled your worker...

Reply:
We can do it later. Just trying to make sure I understand what im reading.